tests/test_modeling_common.py

```diff
@@ -2217,9 +2217,12 @@ def check_device_map_is_respected(self, model, device_map):
     @require_accelerate
     @require_torch_gpu
     def test_disk_offload(self):
+        if all([model_class._no_split_modules is None for model_class in self.all_model_classes]):
+            return
+
         config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
-        if isinstance(getattr(config, "num_hidden_layers", None), int) and config.num_hidden_layers < 5:
-            config.num_hidden_layers = 5
+        if isinstance(getattr(config, "num_hidden_layers", None), int) and config.num_hidden_layers < 4:
+            config.num_hidden_layers = 4
```

Comment:
4 layers is enough for OPT (which was the model that triggered the 5) and it works for GPT-Neo (which wants an even number of layers).

Reply:
I am OK with this as far as it works.

But I feel that this test should be better overwritten by `GPT-Neo`, as it has something special
```
self.attention_layers = self.expand_attention_types_params(attention_types)
```
which causes the failure.

Reply:
There is no need to overwrite for GPT-Neo as it has 4 layers.